app.py

```diff
@@ -1,37 +1,22 @@
 import os
 from logging import getLogger
 from fastapi import FastAPI, Response, status
-from vectorizer import Vectorizer, VectorInput
-from meta import Meta
+from transformer import Transformer, TextInput, TextsInput
 
 
 app = FastAPI()
-vec : Vectorizer
-meta_config : Meta
 logger = getLogger('uvicorn')
 
-
 @app.on_event("startup")
 def startup_event():
-    global vec
-    global meta_config
-
-    cuda_env = os.getenv("ENABLE_CUDA")
-    cuda_support=False
-    cuda_core=""
+    global encoder
+    model_path = "/./models/trans" #we assume the model was already downloaded
+    device = os.environ.get("DEVICE") # device can be any pytorch device. 
+    # Current implementation should detect cuda devices automatically
+    encoder = Transformer().load_model(model_path, device)
 
-    if cuda_env is not None and cuda_env == "true" or cuda_env == "1":
-        cuda_support=True
-        cuda_core = os.getenv("CUDA_CORE")
```

Comment:
The library promises to handle the different cuda configs automatically, needs to be tested in cuda on the cloud.